internal/pkg/docker/dockercompose/decompose.go

```diff
@@ -34,6 +35,15 @@ func DecomposeService(content []byte, svcName string, workingDir string) (*Conve
 		return nil, nil, err
 	}
 
+	// workaround for compose-go automatically loading env files from disk and merging their
+	// content into the "environment" field (the equivalent of the Copilot variables key)
+	// TODO (rclinard-amzn): make a PR to compose-go instead of using this workaround
+	envFiles, err := isolateEnvFiles(service)
+	if err != nil {
+		return nil, nil, fmt.Errorf("isolate env file(s): %w", err)
+	}
+	removeEnvFiles(services)
+
 	project, err := loader.Load(compose.ConfigDetails{
```

Comment:
Do compose-go try to read the env files from the disk here? Sorry I am not sure where that is happening and how we are stopping that here by `isolateEnvFiles`? 
Or we are just overriding it on the [L65](https://github.com/aws/copilot-cli/pull/3935/files#diff-16da4d8aa47f3d1ec247adb442017f524fba30d0f9373942e2635709c223fc65R65)?

Reply:
`compose-go` does try to read the env files from disk and merge them. So we detect the env files for the current service, remove them from all services in the YAML (since compose-go tries to load them for all services), ask compose-go to parse the YAML, and then put them back into the parsed Compose service. Hopefully that makes sense!